xdsl/dialects/riscv.py

```diff
@@ -867,26 +868,22 @@ def custom_print_attributes(self, printer: Printer) -> Set[str]:
         return {"immediate", "rd"}
 
 
-class RdRsIntegerOperation(IRDLOperation, RISCVInstruction, ABC):
+class RdRsOperation(Generic[RDInvT, RSInvT], IRDLOperation, RISCVInstruction, ABC):
     """
     A base class for RISC-V pseudo-instructions that have one destination register and one
     source register.
     """
 
-    rd: OpResult = result_def(IntRegisterType)
-    rs: Operand = operand_def(IntRegisterType)
+    rd: OpResult = result_def(RDInvT)
+    rs: Operand = operand_def(RSInvT)
 
     def __init__(
         self,
         rs: Operation | SSAValue,
         *,
-        rd: IntRegisterType | str | None = None,
```

Comment:
Passing `str` or `None` is a casualty of this change, but in my opinion it's not a great loss to lose this helper functionality. It makes the code a bit more explicit at the callsite when constructing this operation individually, and doesn't have a great impact on the lowering passes due to them calling helpers anyway.